Adjust the test logic on TestGetStaticPodToMirrorPodMap

Chulong-Li opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

The error message for TestGetStaticPodToMirrorPodMap is misleading. The message "correct mirror pod UID" describes the gotMirrorPod.UID, which should be the incorrect UID. Additionally, it currently says that the function returns a UID directly, but it actually returns a pod object map. Fix this grammatical error for clarity.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

Chulong-Li commented:
/sig node

HirazawaUi commented on pkg/kubelet/pod/pod_manager_test.go:
```diff
@@ -303,6 +303,6 @@ func TestGetStaticPodToMirrorPodMap(t *testing.T) {
 	}
 	if gotMirrorPod, ok := m[staticPod]; !ok || gotMirrorPod.UID != mirrorPod.UID {
-		t.Fatalf("GetStaticPodToMirrorPodMap() did not return the correct mirror pod UID %s, wanted mirror pod UID %s", gotMirrorPod.UID, mirrorPod.UID)
+		t.Fatalf("GetStaticPodToMirrorPodMap(): got incorrect mirror pod UID %s, wanted mirror pod UID %s", gotMirrorPod.UID, mirrorPod.UID)
```
After taking a closer look, it seems there’s more than just an grammar error here.

Would this modification make the error message more accurate?

```
gotMirrorPod, ok := m[staticPod]
if !ok {
    t.Fatalf("GetStaticPodToMirrorPodMap(): key %v not found", staticPod)
}
if gotMirrorPod.UID != mirrorPod.UID {
    t.Fatalf("GetStaticPodToMirrorPodMap(): got UID %s, want %s", gotMirrorPod.UID, mirrorPod.UID)
}
```

Chulong-Li replied:
@HirazawaUi Yeah. That logic change makes sense. Thanks for the comment!

Though, I have one suggestion on the first error message. Since, if I understand correctly, "!ok" represents there's no expected mirror pod for a given (and valid) static pod, should we say "mirror pod not found for staticPod %v" for better expression?

```go
gotMirrorPod, ok := m[staticPod]
if !ok {
    t.Fatalf("GetStaticPodToMirrorPodMap(): mirror pod not found for staticPod %v", staticPod)
}
if gotMirrorPod.UID != mirrorPod.UID {
    t.Fatalf("GetStaticPodToMirrorPodMap(): got UID %s, want %s", gotMirrorPod.UID, mirrorPod.UID)
}
```

HirazawaUi replied:
Of course, please modify the PR title after submitting the code to make the description more accurate, as this is no longer just a grammar fix.

Chulong-Li committed:
Fix the logic on TestGetStaticPodToMirrorPodMap

Fix the logic on TestGetStaticPodToMirrorPodMap

Add back the length check logic

HirazawaUi commented:
/ok-to-test

HirazawaUi commented:
This is a small positive change; it’s harmless and makes the unit test output clearer and more accurate logs when it fails.

/lgtm

/assign @SergeyKanzhelev @mrunalp
for approval

Chulong-Li commented:
/assign @tallclair 
Hi @tallclair and @mrunalp, could you please help approve this PR (minor change)? Thanks in advance.

Chulong-Li commented:
Hi @tallclair and @mrunalp, would you like to approve this PR (minor change) if everything looks good? Thanks!

yujuhong commented on pkg/kubelet/pod/pod_manager_test.go:
```diff
@@ -299,10 +299,11 @@ func TestGetStaticPodToMirrorPodMap(t *testing.T) {
 	podManager := NewBasicPodManager()
 	podManager.SetPods([]*v1.Pod{mirrorPod, staticPod, normalPod})
 	m := podManager.GetStaticPodToMirrorPodMap()
-	if len(m) != 1 {
```
A minor thing but the original test also check if the map only contains one mapping. This is lost in the new change.

Chulong-Li replied:
Thanks for the comment. It makes sense. How do you think about the change below with length check logic?
```go
        if len(m) != 1 {
            t.Fatalf("GetStaticPodToMirrorPodMap(): got %d static pods, wanted 1 static pod", len(m))
	}
        gotMirrorPod, ok := m[staticPod]
	if !ok {
            t.Fatalf("GetStaticPodToMirrorPodMap(): mirror pod not found for staticPod %v", staticPod)
	}
	if gotMirrorPod.UID != mirrorPod.UID {
            t.Fatalf("GetStaticPodToMirrorPodMap(): got UID %s, want %s", gotMirrorPod.UID, mirrorPod.UID)
	}
```

Chulong-Li commented:
/retest

yujuhong commented:
/lgtm
/approve

Chulong-Li commented:
@yujuhong Thanks for your review and approval!
